src/Bundle/Tests/Functional/GridUiTest.php

```diff
@@ -214,40 +226,72 @@ public function it_filters_books_by_author_when_an_author_is_used_in_join_in_que
 
         $this->client->request('GET', sprintf('/by-english-authors/books/?criteria[author]=%d', $authorId));
 
-        $this->assertCount(1, $this->getItemsFromCurrentResponse());
-        $this->assertSame('A Study in Scarlet', $this->getFirstItemFromCurrentResponse()['title']);
+        $titles = $this->getBookTitles();
+
+        $this->assertCount(1, $titles);
+        $this->assertSame('A Study in Scarlet', $titles[0]);
     }
 
+    /** @test */
     public function it_includes_all_rows_even_when_sorting_by_a_nullable_path(): void
     {
         $this->client->request('GET', '/authors/');
-        $totalItemsCountBeforeSorting = $this->getTotalItemsCountFromCurrentResponse();
+        $totalItemsCountBeforeSorting = count($this->getAuthorNames());
 
         $this->client->request('GET', '/authors/?sorting[nationality]=desc');
-        $totalItemsCountAfterSorting = $this->getTotalItemsCountFromCurrentResponse();
+
+        $totalItemsCountAfterSorting = count($this->getAuthorNames());
 
         $this->assertSame($totalItemsCountBeforeSorting, $totalItemsCountAfterSorting);
     }
 
-    private function getTotalItemsCountFromCurrentResponse(): int
+    /** @return string[] */
+    private function getBookTitles(): array
```

Comment:
Would be cool to indicate, that it's related to current response/response crawler as it was before.

From cases like this, just be reading code

```php
        $totalItemsCountBeforeSorting = count($this->getAuthorNames());
        $this->client->request('GET', '/authors/?sorting[nationality]=desc');
        $totalItemsCountAfterSorting = count($this->getAuthorNames());
```

it's not obvious, that `getAuthorNames` will return different result.

Reply:
I rename it getAuthorNamesFromResponse, maybe that's better?